src/rgw/rgw_trim_bilog.cc

```diff
@@ -571,13 +735,40 @@ int BucketTrimInstanceCR::operate(const DoutPrefixProvider *dpp)
       return set_cr_error(retcode);
     }
 
+    retcode = maybe_remove_generation();
+    if (retcode < 0) {
+      ldpp_dout(dpp, 4) << "remove removing old generation from log: "
+			<< cpp_strerror(retcode) << dendl;
+      return set_cr_error(retcode);
+    }
+
+    if (clean_info) {
+      yield call(new RGWPutBucketInstanceInfoCR(
+		   store->svc()->rados->get_async_processor(),
+		   store, clean_info->first, false, {},
+		   nullptr, dpp));
```

Comment:
we need to careful about racing writes to this bucket instance, similar to how you did with `retry_raced_bucket_write()`

so we need to pass a valid `objv` here instead of nullptr, and handle `ECANCELED` errors in a retry loop that re-reads the bucket info and checks that the log generation is still there to erase

Reply:
it looks like svc_bucket_sobj.cc is already using `RGWBucketInfo::objv_tracker` for its reads and writes, so we should just need the handling for ECANCELED

Reply:
Done!